src/main/java/com/connexta/search/index/IndexServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.index;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+import com.connexta.search.index.exceptions.IndexException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.Resource;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@AllArgsConstructor
+public class IndexServiceImpl implements IndexService {
+
+  private static final String EXT_EXTRACTED_TEXT = "ext.extracted.text";
+
+  @NotNull private final CrudRepository<Index, String> crudRepository;
+  @NotNull private final WebClient webClient;
+
+  @Override
+  public void index(@NotBlank final String datasetId, @NotNull final URI irmUri)
+      throws IndexException {
```

Comment:
Sonarlint complains about throws for runtime exceptions

Reply:
Ideally we could let them bubble up to a global exception handler that would deal with them. I've added the exceptions in the signature in the past just for readability.